File: TwoSum.js
// /**
//  * @param {number[]} nums
//  * @param {number} target
//  * @return {number[]}
//  */
// var twoSum = function (nums, target) {
//   var targetNumbers = [];
//   for (var i = 0; i < nums.length; i++) {
//     console.log(nums.length);
//     for (var j = nums.length - 1; j > 0; j--) {
//       console.log(nums[i], nums[j]);
//       if (i !== j) {
//         if (nums[i] + nums[j] === target) {
//           return (targetNumbers = [i, j]);
//         }
//       }
//     }
//   }
//   return targetNumbers;
// };

var twoSum = function (nums, target) {
  const hashTable = {};

  for (var i = 0; i < nums.length; i++) {
    let complement = target - nums[i];

    if (hashTable.hasOwnProperty(complement)) {
      return [hashTable[complement], i];
    }

    hashTable[nums[i]] = i;
  }

  return [];
};

var nums = [1, 3, 4, 2];
var target = 6;

console.log(twoSum(nums, target));

File: LeetCodeC#/LeetCodeC#/TwoSum.cs
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> numberDictionary = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int complement = target - nums[i];
            if (numberDictionary.ContainsKey(complement))
            {
                return new int[] { numberDictionary[complement], i };
            }

            if (!numberDictionary.ContainsKey(nums[i]))
            {
                numberDictionary.Add(nums[i], i);
            }
        }

        throw new ArgumentException("No two numbers add up to the target.");
    }

    static void Main()
    {
        int[] nums = { 2, 7, 11, 15 };
        int target = 9;
        Solution solution = new Solution();

        int[] result = solution.TwoSum(nums, target);

        Console.WriteLine("Indices of the two numbers that add up to target:");
        Console.WriteLine($"[{result[0]}, {result[1]}]");
    }
}
